extensions/ql-vscode/src/codeql-cli/cli.ts

```diff
@@ -1750,6 +1763,29 @@ export class CodeQLCliServer implements Disposable {
     this._versionChangedListeners.push(listener);
   }
 
+  private async notifyPackChanged(packDir: string) {
+    const packFilePath = join(packDir, "codeql-pack.yml");
+    if (!existsSync(packFilePath)) {
+      throw new Error(`Pack file ${packFilePath} does not exist`);
+    }
+    await this.languageClient.sendNotification(
+      DidChangeWatchedFilesNotification.type,
+      {
+        changes: [
+          {
+            type: FileChangeType.Changed,
+            uri: Uri.file(packFilePath).toString(),
+          },
+        ],
+      },
+    );
+
+    // restarting the language client has the effect of removing compilation
+    // errors in open ql/qll files that are caused by the pack not having been
+    // installed previously:
+    await this.languageClient.restart();
```

Comment:
Restarting the client is a relatively big hammer to solve this problem. I suspect there might be a cheaper way to accomplish the same thing. But considering how unusual the pack add/pack install operations are and that they take a significant amount of time in the first place — it's probably not the end of the world 🤷

Reply:
The time to restart the language server is much smaller than the time to run most `pack` commands. I think this is fine.

Reply:
Thanks for confirming :) 👍